tempodb/tempodb.go

```diff
@@ -460,23 +467,21 @@ func (rw *readerWriter) cleanMissingTenants(tenants []string) {
 		tenantSet[tenantID] = struct{}{}
 	}
 
+	rw.blockListsMtx.Lock()
```

Comment:
this will probably be fast enough that the simplicity is an improvement.  @dgzlopes do you have any thoughts?

Reply:
we need it for more than that, the internal blocklist maps are unprotected on the very next line (this was causing a data race), and so we need to put the lock around them.